### PR TITLE
[DOCKER-5] create tag pipeline

### PR DESCRIPTION
**Pull request by GZLiew**
- [DOCKER-5] add publish steps to docker hub
- [DOCKER-5] setup docker publish to docker hub
- [DOCKER-5] add plugin debug logging
- [DOCKER-5] add version to monorepo-diff
- [DOCKER-5] update diff command
- [DOCKER-5] remove debug log

[DOCKER-5]: https://benieworldwide.atlassian.net/browse/DOCKER-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCKER-5]: https://benieworldwide.atlassian.net/browse/DOCKER-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCKER-5]: https://benieworldwide.atlassian.net/browse/DOCKER-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCKER-5]: https://benieworldwide.atlassian.net/browse/DOCKER-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCKER-5]: https://benieworldwide.atlassian.net/browse/DOCKER-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOCKER-5]: https://benieworldwide.atlassian.net/browse/DOCKER-5?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .buildkite/tag.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+              secret-uuid: zisch5odkndtzeb6pqupfazww4
+              field: username
+  - label: ":pipeline: triggering deploy pipelines"
+    if: build.tag !~ /^skip:/
+    commands: 
+      - export CONTAINER_NAME=$(echo ${BUILDKITE_TAG} | cut -f1 -d":")
+      - buildkite-agent pipeline upload .buildkite/deploy.yml
+    env:
+      - DOCKER_LOGIN_PASSWORD
+      - DOCKER_LOGIN_USERNAME
```
Comment (smilin-desperado) [inline .buildkite/tag.yml]: **issue:** Multiple steps can run on separate runners so I don't think the env variables exposed in the first step will necessary be available in the second
**suggestion:** Move `plugins:` to step where they're used.

File: .buildkite/tag.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  - label: ":pipeline: triggering custom deploy pipelines"
+    if: build.tag =~ /^skip:\d+\.\d+\.\d+$/
```
Comment (smilin-desperado) [inline .buildkite/tag.yml:8]: **docs:** Might be good to document this if it's a way to skip deploying certain tags

File: .buildkite/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    command:
+      - "docker build -t ${CONTAINER_NAME} -f ./${CONTAINER_NAME}/Dockerfile ."
+  - label: ":docker: Publish to docker hub"
+    depends_on: build
+    command:
+      - "docker tag ${CONTAINER_NAME} tapendium/${BUILDKITE_TAG}"
+      - "docker push tapendium/${BUILDKITE_TAG}"
+    plugins:
+      - docker-login#v2.0.1:
+          username: "${DOCKER_LOGIN_USERNAME}"
```
Comment (smilin-desperado) [inline .buildkite/deploy.yml]: **question:** Does reading from the environment variable work here?

Comment (GZLiew) [reply .buildkite/deploy.yml]: not sure let me try it locally and see

**Review by smilin-desperado — APPROVED** [review]

**Pull request merged**